docs/conceptual/runtime/contracts/ink.md

```diff
@@ -0,0 +1,108 @@
+---
+title: ink!
+---
+
+ink! is a [Rust](https://www.rust-lang.org/)-based embedded domain specific language
+([eDSL](https://wiki.haskell.org/Embedded_domain_specific_language)) for writing
+[Wasm](https://webassembly.org/) smart contracts specifically for the [SRML Contracts module](conceptual/runtime/contracts/contracts_module.md).
+
+## Design
+
+The main goals of ink! are user friendliness, conciseness, and efficiency.
+
+ink! is designed to be as close to the Rust programming language as possible. The language uses
+[attribute macros](https://doc.rust-lang.org/reference/procedural-macros.html#attribute-macros) to
+tag standard Rust structures into understandable contract components.
+
+```rust
+#[ink(...)]
+```
+
+Because ink! follows Rust standards, tools like `rustfmt` and `rust-analyzer` already work out of
+the box.
+
+### Contract Components
+
+ink! should still feel familiar to developers who have programmed using other modern smart contract
+languages. The skeleton of a contract has all of the same components that you might expect:
+
+  * Events - `#[ink(event)]`
+    * Event Topics - `#[ink(topic)]`
+  * Storage - `#[ink(storage)]`
+  * Constructor Functions - `#[ink(constructor)]`
+  * Message Functions - `#[ink(message)]`
+
+### Custom Types
+
+ink! supports custom type definitions needed to interact with the Substrate runtime. By default,
+ink! has access to all the [primitive Substrate types](conceptual/runtime/primitives.md) like
+`AccountId`, `Balance`, `Hash`, etc...
+
+You can specify and interact with additional types which may be specific to your Substrate runtime
+environment.
+
+### Overflow Safety
+
+Being written in Rust, ink! can provide compile-time overflow/underflow safety. Using a Rust
+compiler configuration, you can specify whether you want to support overflowing math, or if you want
+contract execution to panic when overflows occur. No need to continually import "Safe Math"
+libraries, although Rust also provides [integrated checked, wrapped, and saturated math
+functions](https://doc.rust-lang.org/std/primitive.u32.html).
+
+### Test Environment
+
+ink! provides a built in test environment that can be used to perform off-chain unit testing with
+the Rust framework. This makes it simple and easy to ensure that your contract code functions as
+expected, without the need for third party testing platforms.
+
+## ink! vs Solidity
+
+Rust is an ideal smart contract language. It is type safe, memory safe, and free of undefined
+behaviors. It generates small binaries because it doesn’t include extra bloat, like a garbage
+collector, and advanced optimizations and tree shaking remove dead code. Through compiler flags,
+Rust can automatically protect against integer overflow.
```

Comment:
This was told in the last section.

Reply:
I think worth repeating though when comparing to Solidity explicitly right?

Reply:
Same point as above. It hasn't been proved that `overflow-checks = true` on Cargo.toml is doing its job.